cognite/client/_api/datapoints.py

```diff
@@ -1,117 +1,969 @@
+        When passing dict objects with specific parameters, these will take precedence. See examples below.
 
-        Note that you cannot specify the same ids/external_ids multiple times.
+        **Performance hint:**: For better performance and memory usage, consider using `retrieve_arrays(...)` which uses `numpy.ndarrays` for data storage.
 
         Args:
-            start (Union[int, str, datetime]): Inclusive start.
-            end (Union[int, str, datetime]): Exclusive end.
-            id (DatapointsIdMaybeAggregate): Id or list of ids. Can also be object
-                specifying aggregates. See example below.
-            external_id (DatapointsExternalIdMaybeAggregate): External id or list of external
-                ids. Can also be object specifying aggregates. See example below.
-            aggregates (List[str]): List of aggregate functions to apply.
-            granularity (str): The granularity to fetch aggregates at. e.g. '1s', '2h', '10d'.
-            include_outside_points (bool): Whether or not to include outside points.
-            limit (int): Maximum number of datapoints to return for each time series.
-            ignore_unknown_ids (bool): Ignore IDs and external IDs that are not found rather than throw an exception.
+            start (Union[int, str, datetime]): Inclusive start. Default: 1970-01-01 UTC.
+            end (Union[int, str, datetime]): Exclusive end. Default: "now"
+            id (DatapointsIdTypes): Id, dict (with id) or (mixed) list of these. See examples below.
+            external_id (DatapointsExternalIdTypes): External id, dict (with external id) or (mixed) list of these. See examples below.
+            aggregates (List[str]): List of aggregate functions to apply. Default: No aggregates (raw datapoints)
+            granularity (str): The granularity to fetch aggregates at. e.g. '1s', '2h', '10d'. Default: None.
+            limit (int): Maximum number of datapoints to return for each time series. Default: None (no limit)
+            include_outside_points (bool): Whether or not to include outside points. Not allowed when fetching aggregates. Default: False
+            ignore_unknown_ids (bool): Whether or not to ignore missing time series rather than raising an exception. Default: False
 
         Returns:
-            Union[None, Datapoints, DatapointsList]: A Datapoints object containing the requested data, or a list of such objects. If `ignore_unknown_id` is True, single id is requested and it is not found, the function will return `None`.
+            Union[None, Datapoints, DatapointsList]: A `Datapoints` object containing the requested data, or a `DatapointsList` if multiple
+                time series were asked for. If `ignore_unknown_ids` is `True`, a single time series is requested and it is not found, the function
+                will return `None`. The ordering is first ids, then external_ids.
```

Comment:
This note becomes redundant by adding some overloads.

Reply:
Not entirely, `ignore_unknown_ids` may be overridden by an individual dictionary, so I've left it in. I think most people rather read this than look at the 8 overloaded methods anyway 😅